sphinx/ext/autodoc/__init__.py

```diff
@@ -463,9 +462,9 @@ def get_sourcename(self):
         # type: () -> str
         if self.analyzer:
             # prevent encoding errors when the file name is non-ASCII
-            if not isinstance(self.analyzer.srcname, text_type):
-                filename = text_type(self.analyzer.srcname,
-                                     sys.getfilesystemencoding(), 'replace')
+            if not isinstance(self.analyzer.srcname, str):
+                filename = str(self.analyzer.srcname,
+                               sys.getfilesystemencoding(), 'replace')
```

Comment:
@shimizukawa Is this code still needed in py3? I can't imagine the case `srcname` is not `str`.

Reply:
According to the `ModuleAnalyzer.__init__` type definition, `srcname` must be a `str`. Shall I remove this branch then?

https://github.com/sphinx-doc/sphinx/blob/30ec4b6bba3cb256b700ca1d7438ddd01b3e948c/sphinx/pycode/__init__.py#L82-L85